src/riak_kv_pb_bucket.erl

```diff
@@ -96,28 +94,60 @@ encode(Message) ->
     {ok, riak_pb_codec:encode(Message)}.
 
 %% @doc process/2 callback. Handles an incoming request message.
-process(#rpblistbucketsreq{type = Type, timeout=T, stream=S}=Req, State) ->
-    case check_bucket_type(Type) of
-        {ok, GoodType} ->
-            do_list_buckets(GoodType, T, S, Req, State);
-        error ->
-            {error, {format, "No bucket-type named '~s'", [Type]}, State}
+process(#rpblistbucketsreq{type = Type, timeout = T, stream = S} = Req, State) ->
+    Class = case S of
+        true ->
+            {riak_kv, stream_list_buckets};
+        _ ->
+            {riak_kv, list_buckets}
+    end,
+    Accept = riak_core_util:job_class_enabled(Class),
```

Comment:
It seems like there's an opportunity for some significant reduction in code here if riak_core_util:job_class_enabled just took an "Accepted fun" and a "Rejected fun" and wrapped up all the calls to `request_disposition` and `job_class_disabled_message`.  May require an "error format" parameter as well (although it looks here like all the error messages are in binary format?). I assume a similar pattern appears in other PRs like this?